var/spack/repos/builtin/packages/charliecloud/package.py

```diff
@@ -33,6 +34,8 @@ class Charliecloud(AutotoolsPackage):
     depends_on('py-sphinx',           type='build', when='+docs')
     depends_on('py-sphinx-rtd-theme', type='build', when='+docs')
 
+    conflicts('platform=darwin', msg='This package does not build on macOS')
```

Comment:
@adamjstewart would `platform!=linux` work with spack internals?

Reply:
Nope, unfortunately there's no "not" in the spec syntax. You would have to add a similar line for every other possible platform.